hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreUtils.java

```diff
@@ -136,4 +140,26 @@ public static OptionalLong getMaxSequenceIdInList(Collection<HStoreFile> sfs) {
     return largestFile.isPresent() ? StoreUtils.getFileSplitPoint(largestFile.get(), comparator)
         : Optional.empty();
   }
+
+  /**
+   * Returns the configured checksum algorithm.
+   * @param conf The configuration
+   * @return The checksum algorithm that is set in the configuration
+   */
+  public static ChecksumType getChecksumType(Configuration conf) {
+    String checksumName = conf.get(HConstants.CHECKSUM_TYPE_NAME);
```

Comment:
You don't want to do get

return ChecksymType.nameToType(conf.get(HConstants.CHECKSUM_TYPE_NAME, DEFAULT_WHATEVER_IT_IS));

Is this a candidate for StoreContext? (Perhaps if called frequently). Perhaps StoreContext is not available where this is used?

Reply:
will change in next commit. we put it as it's a static helper method, IMO we're writing it in the right Utils class?

Reply:
@saintstack  we have updated it in the last commit, do you mind to have a second look?